test/system/252-quadlet.bats

```diff
@@ -1013,4 +1013,81 @@ EOF
     run_podman rmi $(pause_image)
 }
 
+@test "quadlet - image files" {
+    local quadlet_image_unit=image_test_$(random_string).image
+    local quadlet_image_file=$PODMAN_TMPDIR/$quadlet_image_unit
+    cat > $quadlet_image_file <<EOF
+[Image]
+Image=$IMAGE
+EOF
+
+    # Remove the local image to make sure it will be pulled again
+    run_podman image rm $IMAGE
+
+    # Create a directory for all quadlet files to make sure later steps access previous ones
+    local quadlet_tmpdir=$(mktemp -d --tmpdir=$PODMAN_TMPDIR quadlet.XXXXXX)
+
+    # Have quadlet create the systemd unit file for the image unit
+    run_quadlet "$quadlet_image_file" "$quadlet_tmpdir"
+    # Save the image service name since the variable will be overwritten
+    local image_service=$QUADLET_SERVICE_NAME
+
+    local quadlet_volume_unit=image_test_$(random_string).volume
+    local quadlet_volume_file=$PODMAN_TMPDIR/$quadlet_volume_unit
+    cat > $quadlet_volume_file <<EOF
+[Volume]
+Driver=image
+Image=$quadlet_image_unit
+EOF
+
+    # Have quadlet create the systemd unit file for the image unit
+    run_quadlet "$quadlet_volume_file" "$quadlet_tmpdir"
+    # Save the image service name since the variable will be overwritten
+    local volume_service=$QUADLET_SERVICE_NAME
+    local volume_name=systemd-$(basename $quadlet_volume_file .volume)
+
+    local quadlet_container_unit=image_test_$(random_string).container
+    local quadlet_container_file=$PODMAN_TMPDIR/$quadlet_container_unit
+    cat > $quadlet_container_file <<EOF
+[Container]
+Image=$quadlet_image_unit
+Volume=$quadlet_volume_unit:/vol
+Exec=sh -c "echo STARTED CONTAINER; echo "READY=1" | socat -u STDIN unix-sendto:\$NOTIFY_SOCKET; sleep inf"
+EOF
+
+    # Image should not exist
+    run_podman 1 image exists ${IMAGE}
+    # Volume should not exist
+    run_podman 1 volume exists ${volume_name}
+
+    # Have quadlet create the systemd unit file for the image unit
+    run_quadlet "$quadlet_container_file" "$quadlet_tmpdir"
+    local container_service=$QUADLET_SERVICE_NAME
+
+    service_setup $container_service
+
+    # Image system unit should be active
+    run systemctl show --property=ActiveState "$image_service"
+    assert "$output" = "ActiveState=active" \
+           "volume should be active via dependency"
+
+    # Volume system unit should be active
+    run systemctl show --property=ActiveState "$volume_service"
+    assert "$output" = "ActiveState=active" \
+           "volume should be active via dependency"
+
+    # Image should exist
+    run_podman image exists ${IMAGE}
+
+    # Volume should exist
+    run_podman volume exists ${volume_name}
+
+    run_podman exec $QUADLET_CONTAINER_NAME /bin/sh -c "cat /vol/home/podman/testimage-id"
```

Comment:
Don't need `sh -c` for this. Just: `...exec $blah cat /path`

And, if you choose to accept my `NONLOCAL_IMAGE` suggestion above, this `testimage-id` file will not exist; you'll need a different verification mechanism.

Reply:
Since I moved to using `NONLOCAL` this line was removed altogether. Instead, I use inspect commands to verify that the commands were executed as expected. So, if the volume was created with the image and mounted into the container, I can treat podman itself as a blackbox